File: components/monitoring/sdk/java/src/main/java/io/wafermonitor/monitoring/examples/SimpleJob.java
package io.wafermonitor.monitoring.examples;

import io.wafermonitor.monitoring.*;

/**
 * Simple example of using the Java monitoring SDK.
 */
public class SimpleJob {
    public static void main(String[] args) throws InterruptedException {
        System.out.println("=== Java SDK Simple Job Example ===\n");
        
        // Initialize SDK
        MonitoringSDK.init(
            MonitoringConfig.builder()
                .mode(Mode.SIDECAR)
                .appName("example-java-job")
                .appVersion("1.0.0")
                .siteId("fab1")
                .sidecarUrl("http://localhost:17000")
                .build()
        );
        
        System.out.println("✓ SDK initialized");
        System.out.println("✓ Version: " + MonitoringSDK.getVersion());
        System.out.println();
        
        // Start monitored job
        System.out.println("Starting monitored job...");
        MonitoringContext ctx = MonitoringSDK.start("process-wafer", "JAVA-12345");
        
        // Simulate work with progress updates
        for (int i = 1; i <= 5; i++) {
            Thread.sleep(1000);
            int progress = i * 20;
            String message = String.format("Processing step %d/5", i);
            
            ctx.progress(progress, message);
            System.out.printf("  [%3d%%] %s%n", progress, message);
            
            // Add some metrics
            ctx.addMetric("temperature", 75.5 + i);
            ctx.addMetric("pressure", 1013.25 - i * 0.5);
        }
        
        // Add final metadata
        ctx.addMetadata("operator", "john.doe");
        ctx.addMetadata("machine_id", "WFR-001");
        
        // Finish successfully
        System.out.println("\n✓ Job completed successfully");
        ctx.finish();
        
        // Cleanup
        MonitoringSDK.shutdown();
        System.out.println("✓ SDK shut down\n");
    }
}

File: components/monitoring/sdk/perl/examples/simple_job.pl
#!/usr/bin/env perl

use strict;
use warnings;
use v5.10;
use FindBin;
use lib "$FindBin::Bin/../lib";

use Monitoring;
use Time::HiRes qw(sleep);

say "=== Perl SDK Simple Job Example ===\n";

# Initialize SDK
Monitoring::init(
    mode         => 'sidecar',
    app_name     => 'example-perl-job',
    app_version  => '1.0.0',
    site_id      => 'fab1',
    sidecar_url  => 'http://localhost:17000',
);

say "✓ SDK initialized\n";

# Start monitored job
say "Starting monitored job...";
my $ctx = Monitoring::start('process-data', 'PERL-12345');

# Simulate work with progress updates
for my $i (1..5) {
    sleep(1);
    my $progress = $i * 20;
    my $message = sprintf("Processing step %d/5", $i);
    
    Monitoring::progress($ctx, $progress, $message);
    printf("  [%3d%%] %s\n", $progress, $message);
    
    # Add some metrics
    Monitoring::add_metric($ctx, 'temperature', 75.5 + $i);
    Monitoring::add_metric($ctx, 'pressure', 1013.25 - $i * 0.5);
}

# Add final metadata
Monitoring::add_metadata($ctx, 'operator', 'john.doe');
Monitoring::add_metadata($ctx, 'machine_id', 'WFR-001');

# Finish successfully
say "\n✓ Job completed successfully";
Monitoring::finish($ctx);

# Cleanup
Monitoring::shutdown();
say "✓ SDK shut down\n";

